rosbag2_cpp/test/rosbag2_cpp/test_reindexer.cpp

```diff
@@ -0,0 +1,129 @@
+// limitations under the License.
+//
+// DISTRIBUTION A. Approved for public release; distribution unlimited.
+// OPSEC #4584.
+//
+// Delivered to the U.S. Government with Unlimited Rights, as defined in DFARS
+// Part 252.227-7013 or 7014 (Feb 2014).
+//
+// This notice must appear in all copies of this file and its derivatives.
+
+#include <gmock/gmock.h>
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rcpputils/asserts.hpp"
+#include "rcpputils/filesystem_helper.hpp"
+
+#include "rosbag2_cpp/readers/sequential_reader.hpp"
+#include "rosbag2_cpp/reindexer.hpp"
+
+#include "rosbag2_storage/bag_metadata.hpp"
+#include "rosbag2_storage/metadata_io.hpp"
+#include "rosbag2_storage/topic_metadata.hpp"
+
+#include "mock_converter.hpp"
+#include "mock_converter_factory.hpp"
+#include "mock_metadata_io.hpp"
+#include "mock_storage.hpp"
+#include "mock_storage_factory.hpp"
+
+using namespace testing;  // NOLINT
+
+class ReindexTestFixture : public Test
+{
+public:
+  ReindexTestFixture()
+  {
+    database_path = _SRC_REINDEX_DIR_PATH;
+    target_dir = database_path + "/target_metadata";
+  }
+
+  std::string database_path;
+  std::string target_dir;
+};
+
+TEST_F(ReindexTestFixture, test_multiple_files) {
+  auto bag_dir = database_path + "/multiple_files";
+  std::unique_ptr<rosbag2_cpp::reindexers::Reindexer> reindexer =
+    std::make_unique<rosbag2_cpp::reindexers::Reindexer>();
+
+  rosbag2_storage::StorageOptions so = rosbag2_storage::StorageOptions();
+  so.uri = bag_dir;
+  so.storage_id = "sqlite3";
+
+  reindexer->reindex(so);
+
+  // We should have a metadata file at this point
+  auto generated_file = rcpputils::fs::path(bag_dir + "/metadata.yaml");
+  EXPECT_TRUE(generated_file.exists());
+
+  // We made it here. Excellent. Truly excellent.
+  // Read the generated and target metadata
+  auto metadata_io = std::make_unique<rosbag2_storage::MetadataIo>();
+  auto generated_metadata = metadata_io->read_metadata(bag_dir);
+  auto target_metadata = metadata_io->read_metadata(target_dir + "/multiple_files");
+
+  // Compare versions
+  EXPECT_EQ(generated_metadata.version, target_metadata.version);
+
+  // Compare relative file paths
+  for (const auto & gen_rel_path : generated_metadata.relative_file_paths) {
+    EXPECT_TRUE(
+      std::find(
+        target_metadata.relative_file_paths.begin(),
+        target_metadata.relative_file_paths.end(),
+        gen_rel_path) != target_metadata.relative_file_paths.end());
+  }
+
+  // Compare starting times
+  // Disabled for now, since it may not be possible to 100% recreate
+  //   original starting time from metadata
+  // EXPECT_EQ(generated_metadata.starting_time, target_metadata.starting_time);
+
+  // Compare durations
+  // Disabled for now, since I'm not sure how duraction is created, and if it's correct (jhdcs)
+  // EXPECT_EQ(generated_metadata.duration, target_metadata.duration);
+
+  // Compare message count
```

Comment:
It seems to me that the code here is self-documenting - that's the point of the googletest expectation macro syntax. Double check if you need most of these comments.